lib/spack/spack/build_systems/python.py

```diff
@@ -233,13 +236,42 @@ def update_external_dependencies(self):
             if "python" in self.spec.root:
                 python = self.spec.root["python"]
             else:
-                python = spack.spec.Spec("python")
-                repo = spack.repo.path.repo_for_pkg(python)
-                python.namespace = repo.namespace
-                python._mark_concrete()
-                python.external_path = self.prefix
+                python = self.get_external_python_for_prefix()
+                if not python.concrete:
+                    repo = spack.repo.path.repo_for_pkg(python)
+                    python.namespace = repo.namespace
+                    python._mark_concrete()
+                    python.external_path = self.prefix
             self.spec.add_dependency_edge(python, ("build", "link", "run"))
 
+    def get_external_python_for_prefix(self):
```

Comment:
Needs a docstring. Type hints would also be useful. I have no idea what this function does without documentation, and it's in a file I maintain.

Reply:
The basic idea is to find a python spec that should be associated with an external python depedendent spec. I've added a docstring, let me know if you think the docstring should be more descriptive.

Reply:
Thanks! Can you also add a `Returns:` section and add type hints?